app/src/main/java/ch/epfl/qedit/view/home/HomeQuizListFragment.java

```diff
@@ -62,39 +117,144 @@ private void startQuizActivity(String quizID) {
+                RecyclerView recyclerView,
+                RecyclerView.ViewHolder viewHolder,
+                float dX,
+                float dY,
+                int actionState,
+                boolean isCurrentlyActive) {
+            CustomAdapter.CustomViewHolder customViewHolder =
+                    (CustomAdapter.CustomViewHolder) viewHolder;
+            getDefaultUIUtil()
+                    .onDrawOver(
+                            c,
+                            recyclerView,
+                            customViewHolder.name,
+                            dX,
+                            dY,
+                            actionState,
+                            isCurrentlyActive);
+        }
+
+        @Override
+        public void onSelectedChanged(RecyclerView.ViewHolder viewHolder, int actionState) {
+            if (viewHolder != null) {
+                CustomAdapter.CustomViewHolder customViewHolder =
+                        (CustomAdapter.CustomViewHolder) viewHolder;
+                getDefaultUIUtil().onSelected(customViewHolder.name);
+            }
+        }
+
+        @Override
+        public void onChildDraw(
+                Canvas c,
+                RecyclerView recyclerView,
+                RecyclerView.ViewHolder viewHolder,
+                float dX,
+                float dY,
+                int actionState,
+                boolean isCurrentlyActive) {
+            CustomAdapter.CustomViewHolder customViewHolder =
+                    (CustomAdapter.CustomViewHolder) viewHolder;
+            setVisibilityOnChildDraw(dX, customViewHolder);
+            getDefaultUIUtil()
+                    .onDraw(
+                            c,
+                            recyclerView,
+                            customViewHolder.name,
+                            dX,
+                            dY,
+                            actionState,
+                            isCurrentlyActive);
+        }
+    }
+
+    public class CustomAdapter extends RecyclerView.Adapter {
         private LayoutInflater inflater;
 
-        public CustomAdapter(Context context, ArrayList<Map.Entry<String, String>> entries) {
-            this.entries = entries;
+        public CustomAdapter(Context context) {
             inflater = LayoutInflater.from(context);
         }
 
+        @NonNull
         @Override
-        public int getCount() {
-            return entries.size();
+        public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+            return new CustomViewHolder(inflater.inflate(R.layout.recycler_row, parent, false));
         }
 
         @Override
-        public Object getItem(int position) {
-            return entries.get(position);
+        public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
+            CustomViewHolder customViewHolder = (CustomViewHolder) holder;
+
+            // TODO Find a better way than this...
```

Comment:
There is a TODO here, we don't ask you to fix it, we just remind you in case of you've forgotten it

Reply:
That's because I haven't found better, maybe some have a better idea ?

Reply:
no not really, it is okay i guess